### PR TITLE
Add some pytables3 names for some methods

### PR DESCRIPTION
**Pull request by jeanconn**
Add some pytables3 names for some methods via monkey patching.

### CODE REVIEW

**Comment by jeanconn** [comment]
Is this something of what you had in mind @taldcroft ?  I agree it might tide us over.

File: tables3/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import tables
+from distutils.version import LooseVersion
+
+if LooseVersion(tables.__version__) < LooseVersion('3.0'):
+    # alias some methods to their pytables3 names
+    tables.open_file = tables.openFile
+
+    tables.table.Table.read_where = tables.table.Table.readWhere
+    tables.table.Table.get_where_list = tables.table.Table.getWhereList
+    tables.table.Table.read_where = tables.table.Table.readWhere
```
Comment (jeanconn) [inline tables3/__init__.py]: copy-paste error

**Comment by taldcroft** [comment]
In `/proj/sot/ska/tmp/pt2to3` there is a more comprehensive way to finding the compatibility list.  I started with all repos in `$ska/analysis/dependencies/repos` and grepped for any that have `tables` (`find . -name '*.py' | xargs grep --files-with-match 'import tables'`).  I copied those repos to the `pt2to3` directory and did `git pull origin master` for each.

Then I ran `pt2to3_all.sh` which runs `pt2to3` on every file.  The git diffs then show everything that gets changed from 2 to 3.

Finally `diff_all.sh` shows all the diffs in one go.

This strategy could also be used to actually modify the affected repos for tables 2/3 compatibility.  One could probably come up with an automated way to insert the `import tables3_compat` statement.

**Comment by jeanconn** [comment]
Relatedly, I'm not sure if this is supposed to be called `tables3` , `tables3_api`, or `tables3_compat` at this point.

**Comment by taldcroft** [comment]
I wrote about comment about naming but maybe failed to hit the button.  It should be either `tables3_api` or `tables3_compat`, your choice.  It needs to make clear that this is not a regular package that provides new classes/functions in its namespace.

**Comment by jeanconn** [comment]
Gotcha.  I'd seen that the mica comment had changed and now had `import tables3_api` but wasn't sure about the way you might want to go.  I think I like `tables3_api` and will go with that.

**Comment by jeanconn** [comment]
Regarding "In /proj/sot/ska/tmp/pt2to3 there is a more comprehensive way to finding the compatibility list", yes, I had gotten similar results from tables searches on github.  I still wasn't sure that we really wanted to do this in a more aggregate way for applications that actually write data tables.  As in I figured we'd just update and test their code an application at a time as we transition.

**Comment by taldcroft** [comment]
Just to document our verbal discussion with a concrete example here:

For Ska.engarchive, we could run `pt2to3` on all modules and add the `tables3_api` imports as needed. In flight Ska the code should run exactly the same tables 2.x functions/methods as now and thus doing the update should be a no-op.  Of course we check this with the long regression test.

What is new is that the long regression test might also pass on Ska3 with tables 3.3+.  But note that Ska3 brings along a whole bucket of core package updates, so there is no guarantee things will work.  There are also likely the by-hand API change updates in the 3.x versions of some packages that should be backed out.

**Comment by jeanconn** [comment]
Do you have ideas, @taldcroft , for module tests?

**Comment by taldcroft** [comment]
Update ska.engarchive and run the long test with py2.

**Comment by jeanconn** [comment]
Not sure if you wanted the extended regression test or the ska_testr long test, but both seem to work fine against the new tables3_api branch of Ska.engarchive (for ska_testr I put that as the branch instead of master in test_make_archive_long.sh).

**Comment by taldcroft** [comment]
Can you include the `pt2to3_all.sh` script in the repo (maybe drop the `.sh`?), and then include quick instructions / background on updating a repo or script.  Target audience: ACIS ops or Matt.

**Comment by taldcroft** [comment]
The instructions can be in README.md.

**Comment by jeanconn** [comment]
Enough? https://github.com/sot/tables3_api/blob/first_draft/README.md

**Comment by jeanconn** [comment]
Though I suppose for eng_archive it looks like we're skipping the built-in extended regression tests.  What is the status of that?
https://github.com/sot/eng_archive/blob/ef0e1f6cf33c3765341e5aed9bcf2e0d658cc06b/Ska/engarchive/tests/test_fetch_regr.py#L139

File: tables3_api/tests/test_methods.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import tables
+from .. import __init__ as tables3_api
```
Comment (jeanconn) [inline tables3_api/tests/test_methods.py:2]: @taldcroft is this OK?  Wasn't sure about the convention for this.

Comment (taldcroft) [reply tables3_api/tests/test_methods.py:2]: I noticed this first time around...  I am not sure of a better way without playing around, but it's just in the test so no problem.

**Comment by taldcroft** [comment]
Looks good to me.

**Comment by jeanconn** [comment]
OK, with the README acceptable, I think this code is probably fine for a first release.

**Pull request merged**